storage/src/merkle_storage.rs

```diff
@@ -936,17 +939,25 @@ impl MerkleStorage {
         self.staged.insert(*key, value);
     }
 
-    /// Persists an entry and its descendants from staged area to database on disk.
-    fn persist_staged_entry_to_db(&mut self, entry: &Entry) -> Result<(), MerkleError> {
-        let mut batch: Vec<(EntryHash, ContextValue)> = Vec::new();
-
-        // build list of entries to be persisted
-        self.get_entries_recursively(entry, &mut batch)?;
-
-        // write all entries at once (depends on backend)
-        self.db.put_batch(batch)?;
+    /// Marks all the entries from last commit as used
+    /// so GC can know when to remove them
+    pub fn mark_entries_from_last_commit_as_used(&mut self) -> Result<(), MerkleError> {
```

Comment:
This is probably a good candidate for being moved elsewhere once the "commit event" has been implemented (TE-440). I find it smelly that this logic is in `MerkleStorage`.